substreams/src/store.rs

```diff
@@ -1626,6 +1626,71 @@ macro_rules! impl_delta {
     };
 }
 
+// Returns a `u64` whose high 32 bits are the pointer and low 32 bits are the length.
+fn unpack_ptr_len(packed: u64) -> (*mut u8, u32) {
+    let ptr32 = (packed >> 32) as u32;
+    let len32 = packed as u32;
+    let ptr = ptr32 as usize as *mut u8;
+    (ptr, len32)
+}
+
+pub struct FoundationalStore {
+    store_index: u32
+}
+
+impl FoundationalStore {
+    pub fn new(store_index: u32) -> Self {
+        Self {
+            store_index
+        }
+    }
+
+    pub fn get<K: AsRef<[u8]>>(&self, block_number: u64, key: K) -> Option<GetResponse> {
```

Comment:
I think we should make the API to be `GetResponse` directly, only case where we get `None` if it's called from a non-wasm32 target. Let's just panic for now until we do something for this.